### PR TITLE
fix(virgo): getTextPoint offset calculation

### PR DESCRIPTION
**Pull request by zuozijian3720**
Currently, ZERO_WIDTH_SPACE will count as two cursor position, but it should only count as one.
Fixes: https://github.com/toeverything/blocksuite/issues/2123

### CODE REVIEW

**Review by lawvs — COMMENTED** [review]
Could you add a test for it?

**Comment by zuozijian3720** [comment]
> Could you add a test for it?

Yes, I'll try it

**Comment by Flrande** [comment]
If you find that the local test passes but the pipeline fails, it may be due to abnormal logs that are ignored by the local environment.

File: tests/rich-text.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  await page.waitForTimeout(100);
+  await page.keyboard.down('Shift');
+  await press(page, 'Enter');
+  await page.keyboard.up('Shift');
+  await page.waitForTimeout(100);
+  await type(page, 'asdfg');
+  await page.waitForTimeout(100);
+  await press(page, 'Enter');
+  await page.waitForTimeout(100);
+  await page.keyboard.down('Meta');
```
Comment (lawvs) [inline tests/rich-text.spec.ts:26]: The testing platform is Linux, so there is no Meta key.

You can use `SHORT_KEY` instead, Or you can use the following pattern

`await page.keyboard.press(`${SHORT_KEY}+Enter`, { delay: 50 });`

**Review by lawvs — APPROVED** [review]

**Review by doodlewind — APPROVED** [review]

**Comment by zuozijian3720** [comment]
@lawvs Thanks very much

**Pull request merged**